src/browser/fetch_network.js

```diff
@@ -83,11 +83,19 @@ async function on_data_http(data)
 
         let req_headers = new Headers();
         for(let i = 1; i < headers.length; ++i) {
-            let parts = headers[i].split(/:(.+)/);
-            let key =  parts[0].toLowerCase();
-            let value = parts[1].trim();
+            let parts = headers[i].match(/^([^:]*):(.*)$/);
+            let key = "";
+            let value = "";
+            if(parts !== null && parts.length === 3) {
+                key = parts[1].toLowerCase();
+                value = parts[2].trim();
+            }
+            if(!this.net.validate_header(key, value)) {
+                this.write(new TextEncoder().encode("HTTP/1.1 400 Bad Request\r\nContent-Length: 0"));
+                return;
+            }
```

Comment:
Since we're moving things into a helper function, I'd suggest:

```javascript
           const header = this.net.parse_http_header(headers[i]);
           if(!header) {
                this.write(new TextEncoder().encode("HTTP/1.1 400 Bad Request\r\nContent-Length: 0"));
                return;
            }
```